django/import-export-excel/README.md

```diff
@@ -0,0 +1,160 @@
+
+### When to use this technique?
+
+The data importing will never be more than 2000 rows. (Tip validation and raise
+400 when data is more than 2000 rows)
+
+### How?
+
+This would required a minimal setup, here the guides:
+
+- User upload the excel.
+- API received excel, and validate data inside excel.
+- Import data to the system.
+- When import data to the database try to import in chunks, says you have 2000
+rows of data you split it into 500 rows each to import to the database.
+- Return import status back to user.
+
+Overall process should not take any longer than 30 seconds.
+
+### Architecture
+
+![Simple Architecture](img/simple-architecture.jpeg)
+
+## Advance Technique
+
+### When to use this technique?
+
+You don't know how many rows will be imported to the system at a time.
+
+### How?
+
+This would required more setup because we will need background workers (Celery),
+here are some guides:
+
+- User upload the excel.
+- API received excel, and validate data inside excel.
+- After data validated, send file to background worker, then return status 203 as
+the task has been accepted.
+- When importing to database try to split into chunks, says you have 100,000 of
+rows, you should split into 5,000 each time importing to the database.
+- (Optional) Use more than one worker, so that you can import data in parallels.
+- Use COPY instead of INSERT. There are limitation when using COPY, but COPY is
+way more faster than INSERT. Read more about COPY [here](https://www.postgresql.org/docs/9.2/sql-copy.html).
+
+### Architecture
+
+#### Minimum
+
+![Advance Architecture](img/advance-minimum-architecture.png)
+
+#### Recommended
+
+It's recommended to separate between worker and API server because usually workers
+will use a lot of resources, so if workers down it wouldn't affect API Server.
+
+![Advance Architecture](img/advance-recommended-architecture.png)
+
+# Export Data to Excel
+
+Here are some best practices when working with exporting data from database to
+excel.
+
+## Simple
+
+### When to use this technique?
+
+If the data to export is no more than 2000 rows.
+
+### How?
+
+This would required a minimal setup, here the guides:
+
+- User request excel report via API.
+- API Server received request then query the database.
+- Return excel report via api.
+
+### Architecture
+
+![Simple Architecture](img/simple-architecture.jpeg)
+
+## Advance
+
+### When to use this technique?
+
+If you have no idea how many rows it's gonna be or it will be more than 2000 rows.
+
+### How?
+
+This would required more setup because we will need background workers (Celery),
+here are some guides:
+
+- User request excel file via API.
+- API server received the request, then send this job to background worker,
+finally response 203 status code meaning the job is successfully accepted.
+- In worker, when query data from database it's recommend to split it chunks.
+Says 2000 rows per query, you can use SQL `LIMIT` and `OFFSET` to achieve this.
```

Comment:
I guess we can archive this with Django ORM since I think it is not a best idea to use raw SQL krubb:

https://docs.djangoproject.com/en/3.1/topics/db/queries/#limiting-querysets

Reply:
I think it made it confusing, I want to explicit instruct people to use limit and offset from SQL function. I'm afraid people will select 10,000 row and then split into 2,000 in python. Will update na